### ISSUE TITLE
Add backend selection to Juvix app.

### ISSUE THREAD

**Issue by fdedden** [heliaxdev/juvix#683]
We should be able to select between different backends (Michelson, LLVM Webassembly etc.)

**Comment by cwgoes** [comment]
Yes; there should be a CLI flag. For now the standard libraries will be in part backend-specific; once we have typeclasses we should try to nicely abstract things so that contracts which don't use any backend-specific primitives can compile to any backend.

**Comment by mariari** [comment]
Currently the modules have no notion of what modules they are from. Thus we propose to adding an *alist* like changes to the function name

Currently modules look like:

```juvix
mod function-name arg-1 ... argn = toplevel-1 ... toplevel-n
mod function-name arg-1 ... argn = | guard-1 -> toplevel-11 ... toplevel-1n | guard-n -> toplevel-n1 ... toplevel-nn
```

What the modules could look like

```juvix
mod (function-name :backend backend-name)  arg-1 ... argn = toplevel-1 ... toplevel-n
mod (function-name :backend backend-name) arg-1 ... argn = | guard-1 -> toplevel-11 ... toplevel-1n | guard-n -> toplevel-n1 ... toplevel-nn
```

where all we've added is a key value aspect to the function name

```juvix
:backend backend-name
```

This could be good if we do it in other places as well, as name declarations are always `symbols`, and the form is rather nice to work ovfer.

The syntax of :name name-value is not finalized, and just here for a quick demonstration of what it may look like.

**Comment by cwgoes** [comment]
Hmm, for now I think we should make the backend selection happen per-module, not per-function within a module.

**Comment by mariari** [comment]
> Hmm, for now I think we should make the backend selection happen per-module, not per-function within a module.

This is per module, notice the mod before-hand, I should have said mod-name not function-name....

**Comment by cwgoes** [comment]
Ah I see, looks good then.
